Sources/Commands/GenerateShellScript.swift

```diff
@@ -0,0 +1,328 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Foundation
+import Basic
+import Utility
+
+// todo: support positional arguments; eg generate-shell-script [shell-name]
+
+func generateShellScript(forShell shell: Shell, print: (String) -> ()) {
+    let root = ArgumentParser(usage: "", overview: "Swift compiler")
+    root.add(subparser: "build", parser: SwiftBuildTool(args: []).parser)
+    root.add(subparser: "package", parser: SwiftPackageTool(args: []).parser)
+    root.add(subparser: "test", parser: SwiftTestTool(args: []).parser)
+
+    // Swift compiler flags
+    _ = root.add(option: "-assert-config", kind: String.self, usage: "Specify the assert_configuration replacement. Possible values are Debug, Release, Unchecked, DisableReplacement.")
```

Comment:
I agree that they're helpful but if we're going to do compiler options, we should have it in a separate PR